src/lib/scheduler/task_queue.hpp

```diff
@@ -1,11 +1,14 @@
 #pragma once
 
 #include <stdint.h>
+
 #include <array>
 #include <atomic>
 #include <condition_variable>
 #include <memory>
 
+#include <tbb/concurrent_queue.h>  // NOLINT
+
 #include "concurrentqueue.h"
 #include "lightweightsemaphore.h"
```

Comment:
The semaphore uses some defines of the queue. Add a comment?

Reply:
Ah, got it. Yes, I'd appreciate that

Reply:
Added the category and described the issue in a comment. Further, I forward declared the semaphore to avoid including both the queue and the 

```
#include <tbb/concurrent_queue.h>  // NOLINT(runtime/indentation_namespace): wronlgy identified as a C header.

#include "types.hpp"

namespace moodycamel {  //
class LightweightSemaphore;
}
```

In task_queue.cpp:
```
#include "task_queue.hpp"

#include <memory>
#include <utility>

#include "concurrentqueue.h"  // The lightweight semaphore uses definitions of concurrentqueue.h.
#include "lightweightsemaphore.h"
```

Okay?

Reply:
```
#include <tbb/concurrent_queue.h>  // NOLINT(build/include_order): wronlgy identified as a C header.

#include "concurrentqueue.h"  // The lightweight semaphore uses definitions of concurrentqueue.h.
#include "lightweightsemaphore.h"

#include "types.hpp"
```

Okay?